docs/guide/data_migration.md

```diff
@@ -0,0 +1,160 @@
+# Data migration
+
+You can migrate your Repository data, that is Content items, as well as Content Types, languages, Object states, Sections, etc.,
+between installations using the migration script.
+
+## Exporting data
+
+To export Repository content, use the `ibexa:migrations:generate` command.
+This command generates a YAML file with the requested part of the Repository.
+You can later use this file to import the data.
+
+``` bash
+bin/console ibexa:migrations:generate --type=content --mode=create 
+```
+
+### type
+
+The mandatory `type` parameter defines the type of Repository data to export.
+The following types are available:
+
+- `content`
+- `content_type`
+- `role`
+- `content_type_group`
+- `user`
+- `user_group`
+- `language`
+- `object_state_group`
+- `object_state`
+- `section`
+- `location`
+
+If you do not provide the `type` parameter, the command will ask you to select a type of data.
+
+### mode
+
+The mandatory `mode` parameter defines the action that importing the file will perform.
+The following modes are available:
+
+- `create` - creates new items
+- `update` - updates an existing item
```

Comment:
In this case "update" does not create an item if it does not exist. If an item does not exist before this step runs, it is considered an error and migration will throw = fail.

And yes, it basically merges the definition for the object (depending on the object type properties or content fields can be different). If a field/property is not specified, then it's not considered for update.

Delete option is similar to update in regard to object matching: not being able to find the relevant object results in an exception = failed migration.

> content record exists on both the source and target instance. Will importing the data delete the record from the target? If so - is it intended to remove duplicates, or what whould be a use case here?

In a lot of cases, generating migration files requires some user input / changing migration files afterwards. Migrations are expected to be part of the "source code", i.e. consist of operations performed on all instances. When creating a delete step, you'd expect all instances to perform this command.

You have to understand the use case that we were presented with when we implemented this: it's supposed to be able recreate a full Ibexa platform instance. **Steps should be executed in order**, so in one step a content object could be added, only to be removed later on. Simply removing the content creation step is not enough, because instances that already did perform that step will be left with it.

As a side note, steps performed will be persisted in the instance, so they won't be run twice unless forced.

Reply:
Thank you! It is not me who should understand the process, it is the users. This is why I am asking questions that they might be asking and we could anticipate this by adding the missing pieces to the puzzle.